Public/Src/Cache/ContentStore/DistributedTest/Stores/GrpcCopyContentTests.cs

```diff
@@ -32,12 +32,67 @@ public class GrpcCopyContentTests : TestBase
     {
         private const int FileSize = 1000;
         private const HashType DefaultHashType = HashType.Vso0;
+        private const string LocalHost = "localhost";
         private Context _context;
+        private GrpcCopyClientCache _clientCache;
 
         public GrpcCopyContentTests()
             : base(() => new PassThroughFileSystem(TestGlobal.Logger), TestGlobal.Logger)
         {
             _context = new Context(Logger);
+            _clientCache = new GrpcCopyClientCache(_context);
+        }
+
+        [Fact]
+        public void DuplicateClientsAreTheSameObject()
+        {
+            using (var client1 = _clientCache.Create(LocalHost, 10, true))
+            using (var client2 = _clientCache.Create(LocalHost, 10, true))
+            {
+                Assert.Same(client1, client2);
+            }
+        }
+
+        [Fact]
+        public void ValidateBackgroundCleanup()
+        {
+            var key = new GrpcCopyClientKey(LocalHost, 11, true);
+            using (var client = _clientCache.Create(key.Host, key.GrpcPort, key.UseCompression))
+            { 
+                client._lastUseTime = DateTime.UtcNow - TimeSpan.FromHours(2);
+            }
+
+            _clientCache.StartBackgroundCleanup();
+            var endTime = DateTime.UtcNow + TimeSpan.FromMinutes(1);
+            while (DateTime.UtcNow < endTime)
+            {
+                if (!_clientCache._clientDict.TryGetValue(key, out GrpcCopyClient foundClient))
+                {
+                    return;
+                }
+
+                Task.Delay(1000).GetAwaiter().GetResult();
+            }
+
+            Assert.True(false, $"{nameof(GrpcCopyClient)} was not removed");
+        }
+
+        [Fact]
+        public void IssueSameClientManyTimes()
+        {
+            GrpcCopyClient sameClient = null;
+            for (int i = 0; i < 1000; i++)
+            {
+                using (var client = _clientCache.Create(LocalHost, 42, false))
+                {
+                    if (i == 0)
+                    {
+                        sameClient = client;

```

Comment:
this is fine, but I would suggest to create another client explicitely outside of the loop.